cli/providers/aws/translate.go

```diff
@@ -84,13 +94,21 @@ func loadASG(t *gocf.Template, s *service.Service, e *environment.Environment, a
 
 func userData(containerName, encodedCompose string, e *environment.Environment, signal string) string {
 	uniqueOperationID := uuid.New().String()
+	encodedConfig := ""
+	if e.Docker != nil && e.Docker.Username != "" && e.Docker.Password != "" {
```

Comment:
can we extract this to a function so we can test it? it'll be handy when we add support for custom repositories